core/src/lib.rs

```diff
@@ -328,6 +329,32 @@ impl RigidBody {
     }
 }
 
+/// Component that defines the dominance of the rigid body
+///
+/// When two dynamic rigid bodies collide, the body with the higher dominance will be treated with infinite mass (i.e. will not be pushed).
+/// This is particularly useful in player controller scenarios
+///
+/// It must be inserted on the same entity of a [`RigidBody`]
+///
+/// # Example
+///
+/// ```
+/// # use bevy::prelude::*;
+/// # use heron_core::*;
+/// fn spawn(mut commands: Commands) {
+///     commands.spawn_bundle(todo!("Spawn your sprite/mesh, incl. at least a GlobalTransform"))
+///         .insert(CollisionShape::Sphere { radius: 1.0 }) // Make a body (is dynamic by default)
+///         .insert(RigidBodyDominance {
+///             dominance: i8::MAX, // Define the dominance of the dynamic rigid body. Higher will dominate
+///         });
+/// }
+/// ```
+#[derive(Debug, Component, Copy, Clone, Default, Reflect)]
+pub struct RigidBodyDominance {
+    /// The dominance of the rigid body. The higher value in a collision will dominate
+    pub dominance: i8,
+}
```

Comment:
I think here I would prefer to have the field private and implement `From<i8> for RigidBodyDominance` and `From<RigidBodyDominance> for i8`.

So that we promise a little bit less, and keep the door open to change the internal representation.